### ISSUE TITLE
Reproducibility question: how to reproduce the full training pipeline described in the paper?

### ISSUE THREAD

**Issue by N1cl0ve** [Eric-qi/DiT-IC#1]
Hi, thanks for releasing the code and models.

I am trying to reproduce the full training pipeline described in the paper, rather than only running the released checkpoints. While reading the paper and the repository, I found several mismatches between the paper training description and the current code/configs, so I would like to ask for clarification on the intended reproduction path.

## My goal

I want to reproduce the training procedure described in the paper as closely as possible, including:
- the two-stage training schedule,
- the original LoRA setup,
- the alignment losses,
- and the Stage-2 training strategy.

### 1. Paper training schedule vs README Quick Start do not match

From the paper:
- Stage 1: 100K iterations on `256x256`, batch size `32`
- Stage 2: 60K iterations on `512x512`, batch size `16`

But in the README Quick Start, the example is:
- `train_nogan_ddp.py --config configs/train_256_nogan.yaml`
- `train_ddp.py --config configs/train_256_gan.yaml`

This looks like `256 -> 256`, not `256 -> 512`.

Also, there are `train_512_nogan.yaml` and `train_512_gan.yaml` configs in the repo, so I am not sure which sequence is the correct one for reproducing the paper results.

## 2. Paper LoRA rank vs released configs do not match

The paper states the practical LoRA ranks are:
- VAE decoder rank = `32`
- diffusion transformer rank = `64`

However, the released training configs use:
- `lora_rank_vae: 64`
- `lora_rank_DiT: 128`

The README news also mentions:
- `2026/03/10 Improved perceptual quality by scaling LoRA rank and updated training.`

So I am unsure whether the current configs correspond to:
- the original paper results, or
- an updated post-paper training recipe.

## 3. Stage-2 adversarial warmup in the paper seems not fully reflected in `train_ddp.py`

In the supplementary material, the paper says that in Stage 2:
- adversarial loss is disabled for the first `30%` of iterations,
- and the contrastive co-alignment loss is also only used in the initial `30%`.

However, in the released code:
- `train_nogan_ddp.py` uses `disc_start` in a way that matches a no-GAN warmup,
- but `train_ddp.py` appears to always run generator/discriminator updates from step 0, while relying on `disc_factor` / `disc_start` behavior inside the loss.

This does not seem equivalent to the paper description of “initially disable adversarial loss for 30% of Stage 2”.

## 4. The paper describes latent-text semantic co-alignment, but the released training path does not seem to include it

The paper describes:
- latent-conditioned guidance,
- InternVL-based caption / text semantic supervision,
- and a CLIP-style contrastive co-alignment loss between projected latent embeddings and text embeddings.

But in the code:
- `LatentConditionAlignment` has a contrastive branch,
- yet `use_clip_contrast=False`,
- the dataset loader only reads images,
- and the training scripts do not seem to provide captions/text embeddings.

So I cannot find a complete released path for reproducing the paper’s `L_cond` training component.

## 5. Some loss naming is difficult to map to the paper

There seems to be naming ambiguity in the current implementation:
- in `models/losses.py`, `kl_loss` is actually the rate term from entropy likelihoods,
- while in `models/DiT_IC.py`, another `kl_loss` is used for the self-distillation cosine alignment between denoised latent and encoder latent.

This makes it hard to determine which released code path exactly corresponds to the paper objective.

## Questions

Could you please clarify the following?

1. Which exact configs + commands should be used to reproduce the **paper training pipeline** and the **original paper results**?
2. Are the released configs intended to reproduce the paper, or the updated post-paper recipe mentioned in the README?
3. For the original paper results, should the correct pipeline be:
   - Stage 1: `256 nogan`
   - Stage 2: `512 gan`
   or something else?
4. How should the Stage-2 “first 30% without adversarial loss” be reproduced in the released code?
5. Is the InternVL / caption-based co-alignment training code intentionally omitted from this release, or am I missing the correct training path?
6. If the paper training code is not fully released yet, would it be possible to provide:
   - the exact original configs,
   - the expected training order,
   - and a short note on which parts of the paper objective are not included in the current public repo?

I would really appreciate clarification, because I want to reproduce the training pipeline faithfully rather than mixing the paper setup with the updated release setup.

Thanks.

**Comment by Eric-qi** [comment]
Thank you for your careful reading and thoughtful questions. The updated checkpoints are trained using the same pipeline as in the original paper, with different configuration settings (e.g., LoRA rank, batch size, number of iterations, etc.).

A few clarifications:

1. **Configs:**  
   The provided configs serve as a reference for our updated training recipe. You may create your own config based on the annotated parameters in `config_your_training.yaml` to match your specific needs.

2. **Training pipeline:**  
   - Stage 1: 256 × 256, no GAN  
   - Stage 2 (first ~30%): 512 × 512, no GAN  
   - Stage 2 (remaining): 512 × 512, with GAN  

   In practice, the resolution, batch size, and number of iterations in each stage can be adjusted based on your hardware constraints.

3. **Stage-2 GAN warm-up:**  
   The first 30% of Stage 2 training without adversarial loss is essentially the same as Stage 1. In the current implementation, we provide two options:

   - **Current (more efficient):**  
     Use `train_nogan_ddp.py` for the first 30% (without adversarial loss), which completely removes GAN-related computation, and then switch to `train_ddp.py` for the remaining GAN training.

   - **Alternative (less efficient):**  
     Use `train_ddp.py` only, with `disc_start=xxx`. In this case, the GAN loss is still computed but multiplied by zero before `disc_start`, resulting in unnecessary computational overhead.  
     
     This corresponds to our earlier implementation, and we have kept the interface for compatibility.

4. **Semantic co-alignment:**  
   In our internal training, text embeddings are precomputed and stored offline to avoid calling the vision-language model during training, which reduces memory usage and improves efficiency.  

   The released code retains the relevant interfaces, but the full embedding pipeline is not included. This component is optional, and the training script can run without it.

5. **Loss naming inconsistencies:**  
   This issue comes from legacy design and partial refactoring. Thank you for pointing it out — we will improve the naming clarity in future updates.

6. **Not included in the current release:**  
   - Precomputed text embedding pipeline  
   - Certain inference optimizations (e.g., FP16 inference, tiled inference)  
   - Some engineering utilities used during training  
   - A follow-up variable bitrate version (as fixed-rate models are relatively large for release)

   These components may be released in the future.

Thanks again for your interest and for raising these detailed questions.

**Comment by N1cl0ve** [comment]
Thank you very much for your detailed response and clarification. I sincerely appreciate your help.
